R/parse-body.R

```diff
@@ -192,7 +192,7 @@ parser_multi <- function() {
     toparse <- parse_multipart(value, boundary)
     # content-type detection
     lapply(toparse, function(x) {
-      if (!is.null(x$filename)) {
+      if (x$content_type %in% c(NULL, "application/octet-stream") && !is.null(x$filename)) {
         x$content_type <- getContentType(tools::file_ext(x$filename))
       }
```

Comment:
`c(NULL, "application/octet-stream") == c("application/octet-stream")`

```suggestion
      if (is.null(x$content_type) || isTRUE(x$content_type == "application/octet-stream")) {
        if (!is.null(x$filename)) {
          x$content_type <- getContentType(tools::file_ext(x$filename))
        }
      }
```